tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -180,7 +181,15 @@ StorageManagerCanonical::load_array_schemas_and_fragment_metadata(
   auto filtered_fragment_uris = array_dir.filtered_fragment_uris(
       array_schema_latest.value().get()->dense());
   const auto& meta_uris = array_dir.fragment_meta_uris();
-  const auto& fragments_to_load = filtered_fragment_uris.fragment_uris();
+  auto fragments_to_load = filtered_fragment_uris.fragment_uris();
```

Comment:
We could make a reserve call to fragments_to_load for fragments_to_load.size() + filtered_fragment_uris.fragment_uris_to_vacuum().size() so that emplace_back doesn't generate any memory allocations.

Reply:
.reserve added (tho another instance where that change is not showing up (below) - a little bit ago the change -was- showing, but as I type this is not visible)